packages/frontend/src/features/metricsCatalog/components/MetricTree/index.tsx

```diff
@@ -243,40 +258,35 @@ const MetricTree: FC<Props> = ({ metrics, metricsTree }) => {
     );
 
     const onLayout = useCallback(() => {
-        if (!nodesInitialized || !metricsTree) {
-            return;
-        }
         const layout = getNodeLayout(currentNodes, currentEdges);
 
         setCurrentNodes([...layout.nodes]);
         setCurrentEdges([...layout.edges]);
 
         setLayoutReady(true);
-    }, [
-        nodesInitialized,
-        metricsTree,
-        currentNodes,
-        currentEdges,
-        setCurrentNodes,
-        setCurrentEdges,
-    ]);
+    }, [currentNodes, currentEdges, setCurrentNodes, setCurrentEdges]);
 
     // Runs layout when the nodes are initialized
     useEffect(() => {
-        if (!metricsTree || !nodesInitialized || layoutReady) {
+        if (!nodesInitialized || layoutReady) {
             return;
         }
         onLayout();
-    }, [onLayout, layoutReady, nodesInitialized, metricsTree, fitView]);
+    }, [onLayout, layoutReady, nodesInitialized, fitView]);
 
     // Fits the view when the layout is ready
     useEffect(() => {
         if (nodesInitialized && layoutReady) {
             window.requestAnimationFrame(async () => {
+                // TODO: this second call to onLayout shouldnt be here,
+                // but it was one way I found to make sure layout happens
+                // after nodes are initialized. We could improve this.
+                onLayout();
+                setLayoutReady(false);
```

Comment:
This is not great 😄. Let's call it prototype code. I tried a lot of things, but getting dynamic sized nodes to layout right is messed up. One option is to have static sizes.

Reply:
Yea, actually, this breaks. Marking as a draft.

Reply:
Ok, I made another attempt that seems to work better